NGitLab.Mock/User.cs

```diff
@@ -34,6 +34,8 @@ public User(string userName)
 
         public string WebUrl => Server.MakeUrl(UserName);
 
+        public bool Bot { get; set; }
```

Comment:
I don't think this should be settable. Instead, I would prefer a method such as

````c#
IGitLabClient Project.CreateProjectAccessToken();
````

This way, you can set all properties as indicated in the documentation https://docs.gitlab.com/ee/user/project/settings/project_access_tokens.html#bot-users-for-projects